### DISCUSSION TITLE
Predict with both historical and expected input

### DISCUSSION THREAD

**Discussion by jerronl** [aeon-toolkit/aeon#648] [Q&A] [answered]
When we predict the future, besides the historical data as input, can we also include expected data as input?
For example, following code predicts the pm25 with historical weather conditions:
```
import pandas as pd
import numpy as np
from sklearn.preprocessing import MinMaxScaler,LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.metrics import mean_squared_error
from aeon.regression.deep_learning import InceptionTimeRegressor
from pandas import read_csv

# Load the dataset
dspath='beijing_pm25_data.csv'
data = read_csv(dspath, header=0, index_col=0)
# Preprocess the data (if needed)
# ...


encoder = LabelEncoder()
data['wnd_dir'] = encoder.fit_transform(data['wnd_dir'] )

# Split the data into features (X) and target variable (y)
X = data.values
y = data['pm2.5'].values

# Scale the features and target variable
scaler = MinMaxScaler()
X_scaled = scaler.fit_transform(X)
y_scaled = scaler.fit_transform(y.reshape(-1, 1)).flatten()

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y_scaled, test_size=0.5, shuffle=False)

# Create the RocketRegressor model
model = make_pipeline(InceptionTimeRegressor())

# Train the model
model.fit(X_train, y_train)

# Make predictions on the testing set
y_pred = model.predict(X_test)

# Inverse scale the predictions
y_pred = scaler.inverse_transform(y_pred.reshape(-1, 1)).flatten()

# Calculate the root mean squared error (RMSE)
rmse = np.sqrt(mean_squared_error(y_test, y_pred))

# Print the predictions and RMSE
print("Predictions:", y_pred)
print("RMSE:", rmse)

```

can we also add future weather conditions as input, to see how would the pm25 be in the future if it gets hotter and wetter?

**Comment by jerronl** [comment]
I tried a different approach by padding the future expected value as additional features. Say, we have n features and each of the k sample contains h periods of them. Also we want m expected future control variable. So the X is actually k*(n+m)*h 3D array, which not only not only contains the k*n*h time series, but also almost all zero k*m*h array, with the last period assigned with the expected future value. I tried a little bit and it seems to work. Is it a valid solution or there are better ways?

**Comment by MatthewMiddlehurst** [comment, marked as answer]
This seems like a TSER problem, predicting a value using other time series data rather than predicting the future if I'm correct (seems similar to this data https://zenodo.org/record/3902671)? At least those seem to be the tools you are using currently.

I would say the expected values for each series would just be an extension of the current/historical data, you could just add the expected/forecasted data on the end to create a new dataset? This is not really my area of expertise, and I may be misunderstanding the task here though 🙂.
